gpkit/geometric_program.py

```diff
@@ -1,380 +1,99 @@
-            assert not any([subbed.exp for subbed in subbeds])
-            return np.array([mag(subbed.c) for subbed in subbeds],
-                            np.dtype('float'))
-        else:
-            subbed = p.sub(self["sensitivities"]["variables"],
-                           require_positive=False)
-            assert isinstance(subbed, Monomial)
-            assert not subbed.exp
-            return subbed.c
-
-    def table(self,
-              tables=["cost", "freevariables", "sweptvariables",
-                      "constants", "sensitivities"],
-              fixedcols=True):
-        if isinstance(tables, Strings):
-            tables = [tables]
-        strs = []
-        if "cost" in tables:
-            strs += ["\nCost\n----"]
-            if len(self) > 1:
-                costs = ["%-8.3g" % c for c in self["cost"][:4]]
-                strs += [" [ %s %s ]" % ("  ".join(costs),
-                                        "..." if len(self) > 4 else "")]
-            else:
-                strs += [" %-.4g" % self["cost"]]
-            strs[-1] += unitstr(self.gp.cost.units, into=" [%s] ", dimless="")
-            strs += [""]
-        if "variables" in tables:
-            strs += [results_table(self["variables"],
-                                   "Variables",
-                                   fixedcols=fixedcols)]
-        if "freevariables" in tables:
-            strs += [results_table({k: v for (k, v) in self["variables"].items()
-                                         if k not in self.gp.substitutions
-                                         and k not in self.gp.sweep},
-                                   "Free variables",
-                                   fixedcols=fixedcols)]
-        if "sweptvariables" in tables:
-            strs += [results_table({k: v for (k, v) in self["variables"].items()
-                                         if k in self.gp.sweep},
-                                   "Swept variables",
-                                   fixedcols=fixedcols)]
-        if "constants" in tables:
-            strs += [results_table({k: v for (k, v) in self["variables"].items()
-                                         if k in self.gp.substitutions},
-                                   "Constants",
-                                   fixedcols=fixedcols)]
-        if "sensitivities" in tables:
-            strs += [results_table(self["sensitivities"]["variables"],
-                                   "Constant and swept variable sensitivities",
-                                   fixedcols=fixedcols,
-                                   minval=1e-2,
-                                   printunits=False)]
-        return "\n".join(strs)
-
-
-class GeometricProgram(Model):
-    """Holds a model and cost function for passing to solvers.
+class GeometricProgram(object):
+    """Prepares a collection of posynomials for a GP solver.
```

Comment:
And the docstring could even mention that at this level we do not separate out monomial equality constraints; those are implemented externally vie two inequality constraints. (if my understanding of that is correct).